src/test/tree/DelegationManagerUnit.tree

```diff
@@ -0,0 +1,310 @@
+.
+├── DelegationManager Tree (*** denotes that integration tests are needed to validate path)
+├── when registerAsOperator is called
+│   ├── given that the caller has already registered as operator
+│   │   └── it should revert
+│   ├── it should call `_setOperatorDetails`
+│   │   ├── given that operatorDetails.earningsReceiver is 0 address
+│   │   │   └── it should revert
+│   │   ├── given operatorDetails.stakerOptOutWindowBlocks is > MAX_STAKER_OPT_OUT_WINDOW_BLOCKS
+│   │   │   └── it should revert
+│   │   ├── given caller has already delegated to an operator
+│   │   │   └── it should revert
+│   │   └── it should emit an OperatorDetailsModified event
+│   ├── it should call `_delegate`
+│   │   ├── given that delegation is paused
+│   │   │   └── it should revert
+│   │   ├── it should set the operator delegated to itself and emit a StakerDelegated event
+│   │   ├── given the caller has delegateable shares
+│   │   │   └── it should increase the operator's shares and and emit an OperatorSharesIncreased event
+│   │   └── it should push an operator stake update
+│   └── it should emit an OperatorRegistered event and OperatorMetadataURIUpdated event
```

Comment:
first line should depend on 'given the stakeRegistry address is not the 0 address'
second line seems erroneous / out of place?